metis-processor/src/main/java/eu/europeana/metis/processor/utilities/ImageEnhancerUtil.java

```diff
@@ -13,227 +16,338 @@
+            appendSuccessReport(webResourceWrapper, enhancedImage, thumbnailHex, elapsedTimeInNanoSec, reportRow);
+          } else {
+            appendProcessedReport(webResourceWrapper, thumbnailHex, reportRow);
+          }
+        } else {
+          appendUnProcessedReport(webResourceWrapper, "", reportRow);
         }
+      } catch (MediaExtractionException | IOException e) {
+        LOGGER.error("Failed enhancing thumbnail image {} {}", webResourceWrapper.getAbout(), e);
+        appendFailReport(reportRow);
+      }
+    } else {
+      try {
+        appendDryRunReport(webResourceWrapper, md5Hex(webResourceWrapper.getAbout()), reportRow);
+      } catch (MediaExtractionException e) {
+        LOGGER.error("Failed dry run {} {}", webResourceWrapper.getAbout(), e);
+        appendFailReport(reportRow);
+      }
     }
+  }
 
-    private void appendSuccessReport(WebResourceWrapper thumbnailResource, byte[] enhancedImage, String thumbnailHex, long elapsedTimeInNanoSec, ReportRow reportRow) {
-        reportRow.setHeightBefore(thumbnailResource.getHeight());
-        reportRow.setWidthBefore(thumbnailResource.getWidth());
-        reportRow.setElapsedTime(TimeUnit.MILLISECONDS.convert(elapsedTimeInNanoSec, TimeUnit.NANOSECONDS));
-        reportRow.setImageLinkHex(thumbnailHex);
-        appendNewWidthAndHeight(enhancedImage, reportRow);
-        reportRow.setStatus("SUCCESS");
-        fileCsvImageReporter.appendRow(reportRow);
-    }
+  private void processShownBy(RDF recordToProcess, Aggregation aggregationItem, String recordId) {
+    ReportRow reportRow = new ReportRow();
+    reportRow.setRecordId(recordId);
+    reportRow.setImageLink(aggregationItem.getIsShownBy().getResource());
+    LOGGER.info("{}", aggregationItem.getIsShownBy().getResource());
+    List<WebResourceWrapper> webResourceWrappers = new RdfWrapper(recordToProcess)
+        .getWebResourceWrappers(Set.of(WebResourceLinkType.IS_SHOWN_BY))
+        .stream()
+        .filter(webResourceWrapper -> webResourceWrapper != null && webResourceWrapper.getMimeType() != null)
+        .filter(webResourceWrapper -> webResourceWrapper.getMimeType().startsWith("image/"))
+        .distinct()
+        .collect(Collectors.toList());
+    webResourceWrappers.forEach(webResourceWrapper -> {
+      LOGGER.info("=>SimpleItemShownBy {}", webResourceWrapper.getAbout());
+      processItem(recordId, webResourceWrapper);
+    });
+  }
 
-    private void appendProcessedReport(WebResourceWrapper thumbnailResource, String thumbnailHex, ReportRow reportRow) {
-        reportRow.setHeightBefore(thumbnailResource.getHeight());
-        reportRow.setWidthBefore(thumbnailResource.getWidth());
-        reportRow.setImageLinkHex(thumbnailHex);
-        reportRow.setStatus("ALREADYPROCESSED");
-        fileCsvImageReporter.appendRow(reportRow);
-    }
+  private void processHasView(RDF recordToProcess, Aggregation aggregationItem, String recordId) {
+    ReportRow reportRow = new ReportRow();
```

Comment:
This seems to be unnecessary to be defined here?

Reply:
removed.